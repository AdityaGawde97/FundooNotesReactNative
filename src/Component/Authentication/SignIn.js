import React, { Component } from 'react';
import { View, Text } from 'react-native';
import { styles } from '../../Css/Authentication.style'
import { OutlinedTextField } from 'react-native-material-textfield'
import { Button, CheckBox } from 'react-native-elements';
import { loginUser } from '../../Firebase/AuthServices';

export default class SignIn extends Component {
    constructor(props) {
        super(props);

        this.state = {
            email: '',
            password: '',
            visibleIcon: false,
            load: false,
            errors: {}
        };
    }

    validateForm = () => {
        let errors = {};
        let formIsValid = true;

        if (typeof this.state.email !== "undefined") {
            let pattern = new RegExp(/^(("[\w-\s]+")|([\w-]+(?:\.[\w-]+)*)|("[\w-\s]+")([\w-]+(?:\.[\w-]+)*))(@((?:[\w-]+\.)*\w[\w-]{0,66})\.([a-z]{2,6}(?:\.[a-z]{2})?)$)|(@\[?((25[0-5]\.|2[0-4][0-9]\.|1[0-9]{2}\.|[0-9]{1,2}\.))((25[0-5]|2[0-4][0-9]|1[0-9]{2}|[0-9]{1,2})\.){2}(25[0-5]|2[0-4][0-9]|1[0-9]{2}|[0-9]{1,2})\]?$)/i);
            if (!pattern.test(this.state.email)) {
                formIsValid = false;
                errors["email"] = "*Enter valid email-ID.";
            }
        }

        if (!this.state.email) {
            errors["email"] = "*Enter email id";
            formIsValid = false;
        }

        if (!this.state.password) {
            errors["password"] = "*Enter password";
            formIsValid = false;
        }

        this.setState({
            errors: errors
        })
        setTimeout(()=>this.setState({ load: false }), 1000)
        return formIsValid
    }

    submitUserSignIn = () => {
        this.setState({ load: true })
        
        if (this.validateForm()) {
            loginUser(this.state.email, this.state.password, () => {
                setTimeout(()=>this.setState({ load: false }), 3000)
                this.props.navigation.navigate('Notes')
            }, (errors) => {
                this.setState({
                    errors: errors
                })
            })
        }
    }

    render() {
        return (
            <View style={styles.mainContainer}>

                <Text style={styles.appTitle}>
                    <Text style={{ color: '#4285F4' }}>F</Text>
                    <Text style={{ color: '#DB4437' }}>u</Text>
                    <Text style={{ color: '#F4B400' }}>n</Text>
                    <Text style={{ color: '#4285F4' }}>d</Text>
                    <Text style={{ color: '#0F9D58' }}>o</Text>
                    <Text style={{ color: '#DB4437' }}>o</Text>
                </Text>

                <Text style={styles.createTitle}>
                    Sign in
                </Text>

                <Text
                    style={{
                        flex: 0,
                        textAlign: 'center',
                        fontSize: 17,
                        top: '6%'
                    }}>
                    with your Fundoo Account
                </Text>

                <View style={styles.inputContainer}>

                    <View>
                        <OutlinedTextField
                            testID="testEmail1"
                            label="Email"
                            value={this.state.email}
                            onChangeText={(email) => this.setState({ email })}
                            error={this.state.errors.email}
                            errorColor='red'
                        />
                    </View>

                    <View style={styles.passView2}>
                        <OutlinedTextField
                            testID="testPassword1"
                            label="Password"
                            value={this.state.password}
                            onChangeText={(password) => this.setState({ password })}
                            secureTextEntry={!this.state.visibleIcon}
                            error={this.state.errors.password}
                            errorColor='red'
                        />
                    </View>

                    <View style={{
                        flex: 0, flexDirection: 'row'
                    }}>
                        <CheckBox
                            testID="testPassVisibility"
                            title="Show password"
                            checked={this.state.visibleIcon}
                            onPress={() => this.setState({ visibleIcon: !this.state.visibleIcon })}
                            containerStyle={{
                                backgroundColor: '#fff',
                                borderColor: '#fff',
                                right: 7
                            }}
                        />
                        <Button
                            testID="testForgotPassword"
                            title="forgot password?"
                            type="clear"
                            onPress={() => this.props.navigation.navigate('ForgotPassword')}
                            containerStyle={{ top: 5 }}
                        />
                    </View>

                </View>

                <View
                    style={{
                        flex: 1.8,
                        flexDirection: 'row',
                        justifyContent: 'space-around'
                    }}
                >
                    <Button
                        testID="testCreateAccount"
                        title="Create Account"
                        onPress={() => this.props.navigation.navigate('SignUpPage1')}
                        type="clear"
                    />
                    <View style={{ width: 120 }}>
                        <Button
                            testID="testSignInButton"
                            title="Next"
                            onPress={this.submitUserSignIn}
                            loading={this.state.load}
                        />
                    </View>

                </View>

            </View>
        );
    }
}