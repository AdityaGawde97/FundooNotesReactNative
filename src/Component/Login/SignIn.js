import React, { Component } from 'react';
import { View, Text, Image, TouchableOpacity, ProgressBarAndroid } from 'react-native';
import { styles } from '../../Css/LoginCss'
import { OutlinedTextField } from 'react-native-material-textfield'
import { Button } from 'react-native-elements';
import { loginUser } from '../../Firebase/FirebaseLoginServices';
import Icon from 'react-native-vector-icons/Ionicons'

export default class SignIn extends Component {
    constructor(props) {
        super(props);

        this.state = {
            email: '',
            password: '',
            visibleIcon: false,
            animating: false,
            errors: {}
        };
    }

    validateForm = () => {
        let errors = {};
        let formIsValid = true;

        if (typeof this.state.email !== "undefined") {
            let pattern = new RegExp(/^(("[\w-\s]+")|([\w-]+(?:\.[\w-]+)*)|("[\w-\s]+")([\w-]+(?:\.[\w-]+)*))(@((?:[\w-]+\.)*\w[\w-]{0,66})\.([a-z]{2,6}(?:\.[a-z]{2})?)$)|(@\[?((25[0-5]\.|2[0-4][0-9]\.|1[0-9]{2}\.|[0-9]{1,2}\.))((25[0-5]|2[0-4][0-9]|1[0-9]{2}|[0-9]{1,2})\.){2}(25[0-5]|2[0-4][0-9]|1[0-9]{2}|[0-9]{1,2})\]?$)/i);
            if (!pattern.test(this.state.email)) {
                formIsValid = false;
                errors["email"] = "*Enter valid email-ID.";
            }
        }

        if (!this.state.email) {
            errors["email"] = "*Enter email id";
            formIsValid = false;
        }

        if (!this.state.password) {
            errors["password"] = "*Enter password";
            formIsValid = false;
        }

        this.setState({
            errors: errors
        })

        return formIsValid
    }

    submitUserSignIn = () => {
        if (this.validateForm()) {
            loginUser(this.state.email, this.state.password, () => {
                this.setState({ animating: true })
                alert('You have login successfully')
                //this.props.navigation.navigate('HomeScreen')
            }, (errors) => {
                this.setState({
                    errors: errors
                })
            })
        }
    }

    render() {

        return (
            <View style={styles.mainContainer}>

                <ProgressBarAndroid
                    animating={this.state.animating}
                    styleAttr="Horizontal"
                    color="#2196F3"
                />

                <Text style={styles.appTitle}>
                    <Text style={{ color: '#4285F4' }}>F</Text>
                    <Text style={{ color: '#DB4437' }}>u</Text>
                    <Text style={{ color: '#F4B400' }}>n</Text>
                    <Text style={{ color: '#4285F4' }}>d</Text>
                    <Text style={{ color: '#0F9D58' }}>o</Text>
                    <Text style={{ color: '#DB4437' }}>o</Text>
                </Text>

                <Text style={styles.createTitle}>
                    Sign in
                </Text>

                <Text
                    style={{
                        flex: 0,
                        textAlign: 'center',
                        fontSize: 17,
                        top: '6%'
                    }}>
                    with your Fundoo Account
                </Text>

                <View style={styles.inputContainer}>

                    <View style={styles.inputView}>
                        <OutlinedTextField
                            testID="testEmail1"
                            label="Email"
                            value={this.state.email}
                            onChangeText={(email) => this.setState({ email })}
                            error={this.state.errors.email}
                            errorColor='red'
                        />
                    </View>

                    <View style={styles.passIconViewInSignIn}>

                        <View style={styles.passView2}>
                            <OutlinedTextField
                                testID="testPassword1"
                                label="Password"
                                value={this.state.password}
                                onChangeText={(password) => this.setState({ password })}
                                secureTextEntry={!this.state.visibleIcon}
                                error={this.state.errors.password}
                                errorColor='red'
                            />
                        </View>

                        <View style={{ padding: '3%' }}>
                            <TouchableOpacity
                                testID="testPassIcon1"
                                onPress={() => this.setState({ visibleIcon: !this.state.visibleIcon })}
                            >
                                {
                                    !this.state.visibleIcon ?
                                        <Icon name="md-eye" size={40} /> :
                                        <Icon name="md-eye-off" size={40} />
                                }
                            </TouchableOpacity>
                        </View>
                    </View>

                    <View style={{
                        flex: 1, alignSelf: 'flex-start'
                    }}>
                        <Button
                            testID="testForgotPassword"
                            title="forgot password?"
                            type="clear"
                            onPress={() => this.props.navigation.navigate('ForgotPassword')}
                        />
                    </View>

                </View>

                <View
                    style={{
                        flex: 1.5,
                        flexDirection: 'row',
                        justifyContent: 'space-around'
                    }
                    }
                >
                    <Button
                        testID="testCreateAccount"
                        title="Create Account"
                        onPress={() => this.props.navigation.navigate('SignUpPage1')}
                        type="clear"
                    />
                    <View style={{ width: 120 }}>
                        <Button
                            testID="testSignInButton"
                            title="Next"
                            onPress={this.submitUserSignIn}
                        />
                    </View>

                </View>

            </View>
        );
    }
}