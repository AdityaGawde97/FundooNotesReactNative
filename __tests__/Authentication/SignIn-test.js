import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import SignIn from '../../src/Component/Authentication/SignIn';

const wrapper = shallow(<SignIn/>)

/**************************** SignIn Component Testing ******************************/

it('SignIn Page renders correctly', () => {
    renderer.create(<SignIn />);
});

describe('<SignIn/>', () => {
    it('SignIn Page renders correctly', () => {
        expect(shallow(<SignIn />)).toMatchSnapshot();
    });

    it('checking initial state', () => {
        expect(wrapper.state('email')).toEqual('');
        expect(wrapper.state('password')).toEqual('');
    })

    const emailInput = wrapper.find('OutlinedTextField').first();
    const passInput = wrapper.find('OutlinedTextField').last();

    it('testing state of component after onChangeText event occur', () => {

        emailInput.props().onChangeText('andy@gmail.com');

        passInput.props().onChangeText('Abc@1234');

        expect(wrapper.state('email')).toEqual('andy@gmail.com');
        expect(wrapper.state('password')).toEqual('Abc@1234');
    })

    it('props of OutlinedTextField', () => {
        expect(emailInput.findWhere((node) => node.prop('label') === 'Email')).toExist();
        expect(emailInput.findWhere((node) => node.prop('errorColor') === 'red')).toExist();
    });

    it('testing validateForm mathod for email validation', () => {

        const instance = wrapper.instance()
        expect(instance.validateForm()).toBe(true)

        emailInput.props().onChangeText('andygmail.com');
        expect(instance.validateForm()).toBe(false)

        emailInput.props().onChangeText('andy@gmail.com');
        expect(instance.validateForm()).toBe(true)

    })

    it('testing validateForm mathod for password validation', () => {

        const instance = wrapper.instance()

        passInput.props().onChangeText('');
        expect(instance.validateForm()).toBe(false)

        passInput.props().onChangeText('Abc@1234');
        expect(instance.validateForm()).toBe(true)
    })

    it('testing visible password toggling', () => {
        const showPasswordBtn = wrapper.findWhere((node)=>node.prop('testId') === 'testPassVisibility')
        expect(wrapper.state('visibleIcon')).toEqual(false)

        showPasswordBtn.simulate('press');
        expect(wrapper.state('visibleIcon')).toEqual(true)
        showPasswordBtn.simulate('press');
        expect(wrapper.state('visibleIcon')).toEqual(false)
    })

    it('testing Button', () => {
        expect(wrapper.findWhere((node) => node.prop('title') === 'Create Account').length).toBe(1)
        const nextButton = wrapper.findWhere((node) => node.prop('title') === 'Next')
        wrapper.setState({
            email: 'andgmail.com',
            password: 'bc@1234'
        })

        expect(wrapper.state('email')).toEqual('andgmail.com');
        expect(wrapper.state('password')).toEqual('bc@1234');

        nextButton.simulate('press')
        expect(wrapper.state('errors')).toEqual({
            'email': '*Enter valid email-ID.'
        })
    })
});