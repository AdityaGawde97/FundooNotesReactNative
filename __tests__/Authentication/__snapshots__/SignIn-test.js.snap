// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignIn/> SignIn Page renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "flex": 0,
        "fontFamily": "Roboto",
        "fontSize": 26,
        "top": "4%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#4285F4",
        }
      }
    >
      F
    </Text>
    <Text
      style={
        Object {
          "color": "#DB4437",
        }
      }
    >
      u
    </Text>
    <Text
      style={
        Object {
          "color": "#F4B400",
        }
      }
    >
      n
    </Text>
    <Text
      style={
        Object {
          "color": "#4285F4",
        }
      }
    >
      d
    </Text>
    <Text
      style={
        Object {
          "color": "#0F9D58",
        }
      }
    >
      o
    </Text>
    <Text
      style={
        Object {
          "color": "#DB4437",
        }
      }
    >
      o
    </Text>
  </Text>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "flex": 0,
        "fontSize": 24,
        "top": "5%",
      }
    }
  >
    Sign in
  </Text>
  <Text
    style={
      Object {
        "flex": 0,
        "fontSize": 17,
        "textAlign": "center",
        "top": "6%",
      }
    }
  >
    with your Fundoo Account
  </Text>
  <View
    style={
      Object {
        "flex": 9,
        "flexDirection": "column",
        "margin": "8%",
        "top": "5%",
      }
    }
  >
    <View>
      <OutlinedTextField
        activeLineWidth={2}
        animationDuration={225}
        autoCapitalize="sentences"
        baseColor="rgba(0, 0, 0, .38)"
        disableFullscreenUI={true}
        disabled={false}
        disabledLineType="dotted"
        disabledLineWidth={0.5}
        editable={true}
        errorColor="red"
        fontSize={16}
        label="Email"
        labelFontSize={12}
        lineType="solid"
        lineWidth={1}
        onChangeText={[Function]}
        testID="testEmail1"
        textColor="rgba(0, 0, 0, .87)"
        tintColor="rgb(0, 145, 234)"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      style={
        Object {
          "top": "1.5%",
        }
      }
    >
      <OutlinedTextField
        activeLineWidth={2}
        animationDuration={225}
        autoCapitalize="sentences"
        baseColor="rgba(0, 0, 0, .38)"
        disableFullscreenUI={true}
        disabled={false}
        disabledLineType="dotted"
        disabledLineWidth={0.5}
        editable={true}
        errorColor="red"
        fontSize={16}
        label="Password"
        labelFontSize={12}
        lineType="solid"
        lineWidth={1}
        onChangeText={[Function]}
        secureTextEntry={true}
        testID="testPassword1"
        textColor="rgba(0, 0, 0, .87)"
        tintColor="rgb(0, 145, 234)"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      style={
        Object {
          "flex": 0,
          "flexDirection": "row",
        }
      }
    >
      <Themed.CheckBox
        checked={false}
        containerStyle={
          Object {
            "backgroundColor": "#fff",
            "borderColor": "#fff",
            "right": 7,
          }
        }
        onPress={[Function]}
        testID="testPassVisibility"
        title="Show password"
      />
      <ForwardRef(Themed.Button)
        containerStyle={
          Object {
            "top": 5,
          }
        }
        onPress={[Function]}
        testID="testForgotPassword"
        title="forgot password?"
        type="clear"
      />
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1.8,
        "flexDirection": "row",
        "justifyContent": "space-around",
      }
    }
  >
    <ForwardRef(Themed.Button)
      onPress={[Function]}
      testID="testCreateAccount"
      title="Create Account"
      type="clear"
    />
    <View
      style={
        Object {
          "width": 120,
        }
      }
    >
      <ForwardRef(Themed.Button)
        loading={false}
        onPress={[Function]}
        testID="testSignInButton"
        title="Next"
      />
    </View>
  </View>
</View>
`;
